mysql/Doc_Office.sql

```diff
@@ -1,87 +1,182 @@
+    PRIMARY KEY(Prescription_ID),
+    FOREIGN KEY(doctorid) REFERENCES DOCTOR(Doctor_ID)
 );
 
-DROP TABLE IF EXISTS MEDICAL_TEST;
-CREATE TABLE MEDICAL_TEST(
-	  Test_ID int AUTO_INCREMENT,
-    Doctor_ID int,
-    Test_type varchar(45),
-    Result varchar(100),
-    Patient_SSN char(9),
+DROP TABLE IF EXISTS TEST;
+CREATE TABLE TEST(
+	Test_ID int AUTO_INCREMENT,
+    testname varchar(100),
+    result varchar(100),
+    Prescription_ID int NOT NULL,
     Date_given datetime,
     PRIMARY KEY (Test_ID),
-    FOREIGN KEY (Doctor_ID) REFERENCES DOCTOR(Doctor_ID),
-    FOREIGN KEY (Patient_SSN) REFERENCES PATIENT(SSN)
+    FOREIGN KEY(Prescription_ID) REFERENCES PRESCRIPTION(Prescription_ID)
 );
 
-DROP TABLE IF EXISTS APPOINTMENT;
-CREATE TABLE APPOINTMENT(
-	Appointment_number int,
-    Test_given int,
-	Patient_SSN char(9),
-    Doctor_ID int AUTO_INCREMENT,
-    Date datetime,
-    Room_number int,
-    PRIMARY KEY (Appointment_number),
-    FOREIGN KEY (Patient_SSN) REFERENCES PATIENT(SSN),
-    FOREIGN KEY (Test_given) REFERENCES MEDICAL_TEST(Test_ID),
-    FOREIGN KEY (Doctor_ID) REFERENCES DOCTOR(Doctor_ID)
+DROP TABLE IF EXISTS SPECIALIZATION;
+CREATE TABLE SPECIALIZATION(
+	specialtyid int,
+    specialty varchar(45),
+    PRIMARY KEY (specialtyid)
     );
-
+    
 DROP TABLE IF EXISTS AUDIT;
 CREATE TABLE AUDIT(
-	Doctor_ID int AUTO_INCREMENT,
+	id int AUTO_INCREMENT,
+	Doctor_ID int,
     Action varchar(100),
     Date_modified datetime,
+    PRIMARY KEY (id),
     FOREIGN KEY (Doctor_ID) REFERENCES DOCTOR(Doctor_ID)
     );
+
+DROP TABLE IF EXISTS DOCTORPATIENT;
+CREATE TABLE DOCTORPATIENT (
+  doctorId INT NOT NULL,
+  patientid char(9) NOT NULL,
+  PRIMARY KEY (doctorId, patientid),
+  CONSTRAINT doctorid
+    FOREIGN KEY (doctorId)
+    REFERENCES DOCTOR (Doctor_ID)
+    ON DELETE NO ACTION
+    ON UPDATE NO ACTION,
+  CONSTRAINT patientid
+    FOREIGN KEY (patientid)
+    REFERENCES PATIENT (SSN)
+    ON DELETE NO ACTION
+    ON UPDATE NO ACTION
+);
+
+DROP TABLE IF EXISTS DOCTORTEST;
+CREATE TABLE DOCTORTEST(
+	doctorId int NOT NULL,
+    testid int NOT NULL,
+    PRIMARY KEY (doctorId, testid),
+    CONSTRAINT doctor_id
+	  FOREIGN KEY (doctorId) REFERENCES DOCTOR(Doctor_ID)
+      ON DELETE NO ACTION
+      ON UPDATE NO ACTION,
+	CONSTRAINT testid
+      FOREIGN KEY (testid) REFERENCES TEST(Test_ID)
+      ON DELETE NO ACTION
+      ON UPDATE NO ACTION
+    );
+
+DROP TABLE IF EXISTS DOCTORSPECIALIZATION;
+CREATE TABLE  DOCTORSPECIALIZATION(
+	doctorId int NOT NULL,
+    specialtyid int NOT NULL,
+    PRIMARY KEY (doctorId, specialtyid),
+    CONSTRAINT doctor
+	  FOREIGN KEY (doctorId) REFERENCES DOCTOR(Doctor_ID)
+      ON DELETE NO ACTION
+      ON UPDATE NO ACTION,
+	CONSTRAINT specialtyid
+      FOREIGN KEY (specialtyid) REFERENCES SPECIALIZATION(specialtyid)
+      ON DELETE NO ACTION
+      ON UPDATE NO ACTION
+    );
+
+DROP TABLE IF EXISTS TESTAPPOINTMENT;
+CREATE TABLE  TESTAPPOINTMENT(
```

Comment:
Not sure what this table is for, unless we are supporting multiple tests per appointment